### PR TITLE
chore: YoStarJP operator name ocr fix

### PR DESCRIPTION
**Pull request by Manicsteiner**
before edit
> [2024-01-09 11:29:57.443][TRC][Px2a80][Tx14c8] rec unknown oper: 1 | enter
> [2024-01-09 11:29:57.443][TRC][Px2a80][Tx14c8] asst::BattleHelper::click_oper_on_deployment | enter
> [2024-01-09 11:29:57.496][INF][Px2a80][Tx14c8] Call ` C:\Soft\Android\platform-tools\adb.exe -s emulator-5554 shell input tap 300 1318 ` ret 0 , cost 52 ms , stdout size: 0 , socket size: 0
> [2024-01-09 11:29:57.496][TRC][Px2a80][Tx14c8] ready to sleep 150
> [2024-01-09 11:29:57.650][TRC][Px2a80][Tx14c8] end of sleep 150
> [2024-01-09 11:29:57.650][TRC][Px2a80][Tx14c8] asst::BattleHelper::click_oper_on_deployment | leave, 206 ms
> [2024-01-09 11:29:58.253][INF][Px2a80][Tx14c8] Call ` C:\Soft\Android\platform-tools\adb.exe -s emulator-5554 exec-out "screencap | gzip -1" ` ret 0 , cost 603 ms , stdout size: 6766167 , socket size: 0
> [2024-01-09 11:29:58.318][TRC][Px2a80][Tx14c8] asst::BattleProcessTask::check_in_battle | enter
> [2024-01-09 11:29:58.318][TRC][Px2a80][Tx14c8] asst::BattlefieldMatcher::pause_button_analyze count 375 threshold 200
> [2024-01-09 11:29:58.318][TRC][Px2a80][Tx14c8] asst::BattleProcessTask::check_in_battle | leave, 0 ms
> [2024-01-09 11:29:58.330][TRC][Px2a80][Tx14c8] asst::WordOcr [{ リ-44撃: [ 0, 0, 270, 41 ], score: 0.756618 }] by OCR Rec , cost 9 ms
> [2024-01-09 11:29:58.335][TRC][Px2a80][Tx14c8] Proceed [{ リ-44撃: [ 17, 175, 270, 41 ], score: 0.756618 }]
> [2024-01-09 11:29:58.335][WRN][Px2a80][Tx14c8] ocr with preprocess got a invalid name, try to use detect model
> [2024-01-09 11:29:58.350][TRC][Px2a80][Tx14c8] asst::WordOcr [{ リ-マジ: [ 0, 0, 280, 37 ], score: 0.577258 }] by OCR Pipeline , cost 12 ms
> [2024-01-09 11:29:58.356][TRC][Px2a80][Tx14c8] Proceed [{ リ-マジ: [ 5, 177, 280, 37 ], score: 0.577258 }]
> [2024-01-09 11:29:58.356][ERR][Px2a80][Tx14c8] name is empty
> [2024-01-09 11:29:58.356][TRC][Px2a80][Tx14c8] rec unknown oper: 1 | leave, 912 ms
> [2024-01-09 11:29:58.356][TRC][Px2a80][Tx14c8] rec unknown oper: 4 | enter
> [2024-01-09 11:29:58.356][ERR][Px2a80][Tx14c8] asst::VisionHelper::correct_rect image is empty
> [2024-01-09 11:29:58.368][TRC][Px2a80][Tx14c8] match_templ |  score: 0.966464 rect: [ 505, 630, 60, 60 ] roi: [ 0, 580, 1280, 140 ]
> [2024-01-09 11:29:58.368][TRC][Px2a80][Tx14c8] asst::BattleHelper::click_oper_on_deployment | enter
> [2024-01-09 11:29:58.431][INF][Px2a80][Tx14c8] Call ` C:\Soft\Android\platform-tools\adb.exe -s emulator-5554 shell input tap 1076 1332 ` ret 0 , cost 62 ms , stdout size: 0 , socket size: 0
> [2024-01-09 11:29:58.431][TRC][Px2a80][Tx14c8] ready to sleep 150
> [2024-01-09 11:29:58.594][TRC][Px2a80][Tx14c8] end of sleep 150
> [2024-01-09 11:29:58.594][TRC][Px2a80][Tx14c8] asst::BattleHelper::click_oper_on_deployment | leave, 225 ms
> [2024-01-09 11:29:59.175][INF][Px2a80][Tx14c8] Call ` C:\Soft\Android\platform-tools\adb.exe -s emulator-5554 exec-out "screencap | gzip -1" ` ret 0 , cost 580 ms , stdout size: 6600365 , socket size: 0
> [2024-01-09 11:29:59.239][TRC][Px2a80][Tx14c8] asst::BattleProcessTask::check_in_battle | enter
> [2024-01-09 11:29:59.239][TRC][Px2a80][Tx14c8] asst::BattlefieldMatcher::pause_button_analyze count 375 threshold 200
> [2024-01-09 11:29:59.239][TRC][Px2a80][Tx14c8] asst::BattleProcessTask::check_in_battle | leave, 0 ms
> [2024-01-09 11:29:59.260][TRC][Px2a80][Tx14c8] asst::WordOcr [{ リード4印へr: [ 0, 0, 270, 41 ], score: 0.549078 }] by OCR Rec , cost 18 ms
> [2024-01-09 11:29:59.264][TRC][Px2a80][Tx14c8] Proceed [{ リード4印へr: [ 17, 175, 270, 41 ], score: 0.549078 }]
> [2024-01-09 11:29:59.264][WRN][Px2a80][Tx14c8] ocr with preprocess got a invalid name, try to use detect model
> [2024-01-09 11:29:59.289][TRC][Px2a80][Tx14c8] asst::WordOcr [{ リード。4ドシ: [ 0, 0, 280, 37 ], score: 0.635791 }] by OCR Pipeline , cost 22 ms
> [2024-01-09 11:29:59.293][TRC][Px2a80][Tx14c8] Proceed [{ リード。4ドシ: [ 5, 177, 280, 37 ], score: 0.635791 }]
> [2024-01-09 11:29:59.293][ERR][Px2a80][Tx14c8] name is empty
> [2024-01-09 11:29:59.293][TRC][Px2a80][Tx14c8] rec unknown oper: 4 | leave, 937 ms

after edit

> [2024-01-09 11:32:47.231][TRC][Px3018][Tx41ac] asst::WordOcr [{ リ-44撃: [ 0, 0, 270, 41 ], score: 0.756618 }] by OCR Rec , cost 9 ms
> [2024-01-09 11:32:47.233][TRC][Px3018][Tx41ac] Proceed [{ 老鲤: [ 17, 175, 270, 41 ], score: 0.756618 }]
> [2024-01-09 11:32:47.233][TRC][Px3018][Tx41ac] asst::AvatarCacheManager::set_avatar | enter
> [2024-01-09 11:32:47.233][INF][Px3018][Tx41ac] asst::AvatarCacheManager::set_avatar 老鲤 , overlay: 1
> [2024-01-09 11:32:48.161][TRC][Px3018][Tx41ac] asst::WordOcr [{ リード4印へr: [ 0, 0, 270, 41 ], score: 0.549078 }] by OCR Rec , cost 35 ms
> [2024-01-09 11:32:48.164][TRC][Px3018][Tx41ac] Proceed [{ 苇草: [ 17, 175, 270, 41 ], score: 0.549078 }]
> [2024-01-09 11:32:48.164][TRC][Px3018][Tx41ac] asst::AvatarCacheManager::set_avatar | enter
> [2024-01-09 11:32:48.164][INF][Px3018][Tx41ac] asst::AvatarCacheManager::set_avatar 苇草 , overlay: 1

部分名称较短的干员在名称ocr时会混入其他字符，并且得分较低。或许可以通过限制得分来改进？

### CODE REVIEW

File: resource/global/YoStarJP/resource/tasks.json [hunk 1]
```diff
@@ -649,7 +649,7 @@
                 "风笛"
             ],
             [
-                "^リード$",
+                "^リード",
```
Comment (zzyyyl) [inline resource/global/YoStarJP/resource/tasks.json:652]: 这玩意不能这么改吧，焰苇的日文名是啥

Comment (zzyyyl) [reply resource/global/YoStarJP/resource/tasks.json:652]: cc63deefc00028590ac7d0f87ca6f5c49cb5eede

Comment (zzyyyl) [reply resource/global/YoStarJP/resource/tasks.json:652]: ！居然是 "焔影リード"，那没事了

**Comment by Manicsteiner** [comment]
发现了更多干员名称ocr识别出现了问题，可能是 #7773 中修改了ocr识别范围导致的.
已知："^サガ", "嵯峨"

**Comment by Manicsteiner** [comment]
移除了BattleOperName的roi修改后虽然好了几个，但整体还是有问题，感觉还是玄学问题

> [2024-01-09 23:43:17.177][TRC][Px2d10][Tx2844] asst::WordOcr [{ メィー-7（ン: [ 0, 0, 181, 34 ], score: 0.697225 }] by OCR Rec , cost 9 ms
> [2024-01-09 23:43:17.181][TRC][Px2d10][Tx2844] Proceed [{ メィー-7（ン: [ 17, 175, 181, 34 ], score: 0.697225 }]
> [2024-01-09 23:43:18.178][TRC][Px2d10][Tx2844] asst::WordOcr [{ プラチナへ: [ 0, 0, 181, 31 ], score: 0.962592 }] by OCR Rec , cost 9 ms
> [2024-01-09 23:43:18.182][TRC][Px2d10][Tx2844] Proceed [{ プラチナへ: [ 17, 185, 181, 31 ], score: 0.962592 }]
> asst::BattleProcessTask::check_in_battle | leave, 0 ms
> [2024-01-09 23:43:19.143][TRC][Px2d10][Tx2844] asst::WordOcr [{ スワイヤーの: [ 0, 0, 181, 32 ], score: 0.899042 }] by OCR Rec , cost 11 ms
> [2024-01-09 23:43:19.150][TRC][Px2d10][Tx2844] Proceed [{ スワイヤーの: [ 17, 184, 181, 32 ], score: 0.899042 }]
> [2024-01-09 23:43:20.077][TRC][Px2d10][Tx2844] asst::WordOcr [{ スペクタム: [ 0, 0, 170, 31 ], score: 0.908175 }] by OCR Rec , cost 10 ms
> [2024-01-09 23:43:20.083][TRC][Px2d10][Tx2844] Proceed [{ スペクタム: [ 17, 178, 170, 31 ], score: 0.908175 }]

**Comment by Constrat** [comment]
@Manicsteiner the issue happens with the OperBox or "in-fight".
If it's in-fight the ROI might not be exactly correct. You should try "picking" the operator with the longest name to give a "maximum" ROI. ファイヤーホイッスル (Firewhistle) seems the longest name I could fine. You can test with that.

Remember that some items might have longer names compared to the operators.

**Comment by Manicsteiner** [comment]
In-fight. The longest name in normal battle should be "白铁™多功能平台"/“Stainless™ Multifunctional platform”/“ステインレス™多機能プラットフォーム”, it even needs a larger ROI.
Screenshot_1704853492
I think a better way to fix it issue is not to limit the ending characters but to limit the subsequent characters not to overlap with the names of other operators, and only fix the problems of commonly used operators, ignoring the problems of a few infrequently used operators.

**Comment by Manicsteiner** [comment]
暂时可以了
永久遗留问题：干员“梅/メイ”无法被识别，干员"白铁/ステインレス"的召唤物无法被识别

**Comment by Constrat** [comment]
@Manicsteiner A suggestion regarding the ROI. If you want to / have time. You could "calculate" the median value of the operators chars number. Use that one. For operators that go over it, just add `.*`.

**Comment by Manicsteiner** [comment]
最后一个世纪难题
稀音/シーン

> [2024-01-10 23:15:47.197][TRC][Px32dc][Tx07d4] asst::WordOcr [{ シージ-ト~: [ 0, 0, 270, 41 ], score: 0.753006 }] by OCR Rec , cost 9 ms
> [2024-01-10 23:15:47.201][TRC][Px32dc][Tx07d4] Proceed [{ 推进之王: [ 17, 175, 270, 41 ], score: 0.753006 }]

推进之王/シージ

> [2024-01-10 23:15:48.111][TRC][Px32dc][Tx07d4] asst::WordOcr [{ シージて休: [ 0, 0, 270, 41 ], score: 0.746087 }] by OCR Rec , cost 9 ms
> [2024-01-10 23:15:48.115][TRC][Px32dc][Tx07d4] Proceed [{ 推进之王: [ 17, 175, 270, 41 ], score: 0.746087 }]

理论上这个ocr结果还和皮肤有关，实在想不出什么区分方法了

**Comment by Constrat** [comment]
it seems like for the Japanese OCR some recognition parameters might need to be changed. As of now, increasing the ROI to allow longer texts is definitely the best approach (as proved for the YostarEN client), the remaining issue is that the background images (operator sprites) for the OCR in Japanese make characters appear, while in EN I pretty much never found this issue.

iirc, in both BattleOperName and the SSS counterpart, in MaaCore the text is modified, through threshold and expansion. Maybe we could pass them as specialParameters and tweak them for YostarJP

**Pull request merged**